src/runner_manager.py

```diff
@@ -0,0 +1,358 @@
+
+        # Clean up offline runners
+        offline_runners = [r for r in runners if not r.online]
+        if offline_runners:
+            runner_names = ", ".join(r.name for r in offline_runners)
+            logger.info("Cleaning up offline runners: %s", runner_names)
+
+            for runner in offline_runners:
+                runner.remove()
+
+        # Add/Remove runners to match the target quantity
+        online_runners = [r for r in runners if r.exist and r.online]
+        delta = quantity - len(online_runners)
+        if delta > 0:
+            logger.info("Getting registration token for GitHub runners.")
+            if isinstance(self.path, GitHubRepo):
+                token = self._github.actions.create_registration_token_for_repo(
+                    owner=self.path.owner, repo=self.path.repo
+                )
+            else:
+                token = self._github.actions.create_registration_token_for_org(org=self.path.org)
+
+            logger.info("Adding %i additional runner(s)", delta)
+            for _ in range(delta):
+                runner = Runner(
+                    self._github,
+                    self._jinja,
+                    self._lxd,
+                    self.app_name,
+                    self.path,
+                    self.proxies,
+                    self._generate_runner_name(),
+                )
+                runner.create(
+                    self.image, resources, self.runner_bin_path, self.reconcile_interval, token
+                )
+        elif delta < 0:
+            idle_runners = [r for r in online_runners if not r.busy]
+            offset = min(-delta, len(idle_runners))
+            if offset != 0:
+                logger.info("Removing %i runner(s)", offset)
+                remove_runners = idle_runners[:offset]
+                runner_names = ", ".join(r.name for r in remove_runners)
+                for runner in remove_runners:
+                    runner.remove()
+            else:
+                logger.info("There are no idle runner to remove.")
+
+        return delta
+
+    def clean(self) -> int:
+        """Remove existing runners.
+
+        Returns:
+            Number of runner removed.
+        """
+
+        runners = [r for r in self._get_runners() if r.exist]
+        runner_names = ", ".join(r.name for r in runners)
+        logger.info("Removing existing local runners: %s", runner_names)
+
+        for runner in runners:
+            runner.remove()
+
+        return len(runners)
+
+    def _generate_runner_name(self) -> str:
+        """Generate a runner name based on charm name.
+
+        Returns:
+            Generated name of runner.
+
+        TODO:
+            Consider name collusion. LXD would fail to create VM on name collusion.
+        """
+        # Generated a suffix for naming propose, not used as secret.
+        suffix = "".join(random.choices(string.ascii_lowercase + string.digits, k=6))  # nosec B311
```

Comment:
what about `uuid`, would that be useful here?

Reply:
It will solve the problem, but the name would be really long (and may cause problems).

Reply:
What kind of problems?